### PR TITLE
fix: detach charges and cell in PME reciprocal virial background corr…

### PR DESCRIPTION
**Pull request by zubatyuk**
# ALCHEMI Toolkit-Ops Pull Request

## Description

When `hybrid_forces=True`, the reciprocal-space background virial correction in `_pme_reciprocal_space_impl` used the original `charges` and `cell` tensors instead of the detached. This caused the returned virial to retain a `grad_fn` when charges have `requires_grad=True`, violating the documented "forward-only" contract of the hybrid forces mode.

Thanks @vsimkus for reporting.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Performance improvement
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] CI/CD or infrastructure change

## Related Issues

Relates to #57 (feat: Add hybrid forces mode for PME/Ewald)

## Changes Made

- In `_pme_reciprocal_space_impl` (`nvalchemiops/torch/interactions/electrostatics/pme.py`), replaced `charges` with `chg_spline` and `cell` with `cell_spline` in the background virial correction block (both batch and single-system paths). These variables are identical when `hybrid_forces=False` and are the `.detach()`-ed copies when `hybrid_forces=True`, so no behavioral change occurs outside of hybrid mode.
- Added `test_hybrid_virial_detached_with_grad_charges` to `TestPMEHybridForces` in `test/interactions/electrostatics/bindings/torch/test_pme.py`. The test creates charges via `torch.nn.Linear` so they carry `requires_grad=True`, calls `pme_reciprocal_space` with `hybrid_forces=True` and `compute_virial=True`, and asserts the virial has no `grad_fn`.

## Testing

- [x] Unit tests pass locally (`make pytest`)
- [x] Linting passes (`make lint`)
- [x] New tests added for new functionality meets coverage expectations?

## Checklist

- [x] I have read and understand the [Contributing Guidelines](../CONTRIBUTING.md)
- [ ] I have updated the [CHANGELOG.md](../CHANGELOG.md)
- [x] I have performed a self-review of my code
- [ ] I have added docstrings to new functions/classes
- [ ] I have updated the documentation (if applicable)

## Additional Notes

The existing `test_hybrid_virial_forward_only` test did not catch this bug because `create_dipole_system` returns charges with `requires_grad=False`. The `charges.sum()` operation on a leaf tensor without grad produces no `grad_fn`, so the `v_hyb.grad_fn is None` assertion passed vacuously. The new test explicitly uses charges that flow through an `nn.Linear` to reproduce the issue.

> [!TIP]
> This repository uses Greptile, an AI code review service, to help conduct
> pull request reviews. We encourage contributors to read and consider suggestions
> made by Greptile, but note that human maintainers will provide the necessary
> reviews for merging: Greptile's comments are **not** a qualitative judgement
> of your code, nor is it an indication that the PR will be accepted/rejected.
> We encourage the use of emoji reactions to Greptile comments, depending on
> their usefulness and accuracy.

### CODE REVIEW

File: test/interactions/electrostatics/bindings/torch/test_pme.py [hunk 1]
```diff
@@ -3939,6 +3939,35 @@ def test_hybrid_virial_forward_only(self, device):
...
+            cell,
+            alpha=0.3,
+            mesh_dimensions=(16, 16, 16),
+            compute_forces=True,
+            compute_virial=True,
+            hybrid_forces=True,
+        )
+
+        assert charges.requires_grad is True
+        assert v_hyb.grad_fn is None
```
[the remark this answers is not in the document]
Comment (dallasfoster) [reply test/interactions/electrostatics/bindings/torch/test_pme.py]: I think this is actually a good test to have.

[the remark this answers is not in the document]
Comment (zubatyuk) [reply test/interactions/electrostatics/bindings/torch/test_pme.py]: Extended this test in da827c70b6cf42c817db3768ecde243674455ad6

**Review by dallasfoster — APPROVED** [review]

**Comment by dallasfoster** [comment]
/ok to test [da827c7](https://github.com/NVIDIA/nvalchemi-toolkit-ops/pull/65/commits/da827c70b6cf42c817db3768ecde243674455ad6)
